### ISSUE TITLE
  AttributeError: 'Legend' object has no attribute 'draggable'

### ISSUE THREAD

**Issue by jpvilla** [semiautomaticgit/SemiAutomaticClassificationPlugin#63]
WARNING    Traceback (most recent call last):
              File "/Users/Juan1/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/SemiAutomaticClassificationPlugin/semiautomaticclassificationplugin.py", line 1296, in newProjectLoaded
              cfg.spSigPlot.refreshPlot()
              File "/Users/Juan1/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/SemiAutomaticClassificationPlugin/spectralsignature/spectralsignatureplot.py", line 463, in refreshPlot
              self.signaturePlot()
              File "/Users/Juan1/Library/Application Support/QGIS/QGIS3/profiles/default/python/plugins/SemiAutomaticClassificationPlugin/spectralsignature/spectralsignatureplot.py", line 711, in signaturePlot
              cfg.uisp.Sig_Widget.sigCanvas.ax.legend(pL, pLN, bbox_to_anchor=(0.1, 0.0, 1.1, 1.0), loc=1, borderaxespad=0.).draggable()
             AttributeError: 'Legend' object has no attribute 'draggable'

**Comment by jpvilla** [comment]
como soluciono ese error 
Gracias

**Comment by LittleWhite-tb** [comment]
Hello,

Got the same error on Linux Manjaro, with QGIS 3.80 and matplotlib 3.1.1-1.
It happens certainly only with matplotlib 3.1.1, cause they changed the draggable() function to set_dragabble() on Legend. (Source: https://matplotlib.org/api/api_changes.html#removals).

**Comment by LittleWhite-tb** [comment]
Can be quick fixed by editing the script yourself:
Edit the spectralsignatureplot.py file, go to line 711 and change it for:
```
cfg.uisp.Sig_Widget.sigCanvas.ax.legend(pL, pLN, bbox_to_anchor=(0.1, 0.0, 1.1, 1.0), loc=1, borderaxespad=0.).set_draggable(True)
```

**Comment by semiautomaticgit** [comment]
Thank you. 
It should be fixed in SCP 6.3.1
